jm-glass/src/ChordPage.tsx

```diff
@@ -0,0 +1,99 @@
+/** @jsxImportSource @emotion/react */
+import React from 'react';
+import { CHORDS, StringedInstrument } from 'jmusic-engine';
+import { Chord, ChordSelector, NoteSelector } from './components';
+import {
+  ChordContainer,
+  Header,
+  HeaderContainer,
+  Logo,
+  Page,
+  PagesBar,
+  PagesContainer,
+} from '@design/Theme';
+
+const CACHE_HOST = '/jmusic/cache';
+const guitar = StringedInstrument.guitar;
+
+interface SimplifiedChord {
+  notes: number[];
+  playability: number;
+  inversion: number;
+}
+
+function ChordPage() {
+  const [state, stateSet] = React.useState<{
+    note: number;
+    chord: number;
+  }>({ note: 4, chord: 0 });
+  const [groups, groupsSet] = React.useState<SimplifiedChord[][]>([]);
+  const [group, groupSet] = React.useState<number>(0);
+
+  React.useEffect(() => {
+    const getChords = async () => {
+      const req = await fetch(
+        `${CACHE_HOST}/guitar_Guitar Standard_${CHORDS[state.chord].name}_${
+          state.note
+        }.json`
+      );
+      const resp = await req.json();
+      const newChords: SimplifiedChord[] = resp.chords;
+      groupSet(0);
+      const newGroups = [];
```

Comment:
Suggesting this as a nice to do.
It might be better not to mutate the array here and instead use a transformation function.
Ramda can help to make these tasks easy.
Here is the same transformation with Ramda. (using Ramda playground).

https://ramdajs.com/repl/?v=0.27.0#?const%20chords%20%3D%20%5B%7B%22notes%22%3A%5B0%2C2%2C2%2C1%2C0%2C0%5D%2C%22inversion%22%3A1%2C%22playability%22%3A1%2C%22extras%22%3A3%7D%2C%7B%22notes%22%3A%5B7%2C7%2C6%2C4%2C5%2C4%5D%2C%22inversion%22%3A3%2C%22playability%22%3A1%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B7%2C7%2C9%2C9%2C9%2C7%5D%2C%22inversion%22%3A3%2C%22playability%22%3A1%2C%22extras%22%3A3%7D%2C%7B%22notes%22%3A%5B12%2C11%2C9%2C9%2C9%2C12%5D%2C%22inversion%22%3A1%2C%22playability%22%3A1%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B12%2C14%2C14%2C13%2C12%2C12%5D%2C%22inversion%22%3A1%2C%22playability%22%3A1%2C%22extras%22%3A3%7D%2C%7B%22notes%22%3A%5B19%2C19%2C18%2C16%2C17%2C16%5D%2C%22inversion%22%3A3%2C%22playability%22%3A1%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B19%2C19%2C21%2C21%2C21%2C19%5D%2C%22inversion%22%3A3%2C%22playability%22%3A1%2C%22extras%22%3A3%7D%2C%7B%22notes%22%3A%5B4%2C7%2C6%2C4%2C5%2C4%5D%2C%22inversion%22%3A2%2C%22playability%22%3A2%2C%22extras%22%3A1%7D%2C%7B%22notes%22%3A%5B16%2C19%2C18%2C16%2C17%2C16%5D%2C%22inversion%22%3A2%2C%22playability%22%3A2%2C%22extras%22%3A1%7D%2C%7B%22notes%22%3A%5B12%2C11%2C14%2C13%2C12%2C12%5D%2C%22inversion%22%3A1%2C%22playability%22%3A3%2C%22extras%22%3A1%7D%2C%7B%22notes%22%3A%5B7%2C11%2C9%2C9%2C9%2C7%5D%2C%22inversion%22%3A3%2C%22playability%22%3A4%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B4%2C2%2C2%2C1%2C0%2C0%5D%2C%22inversion%22%3A2%2C%22playability%22%3A5%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B4%2C2%2C2%2C4%2C5%2C4%5D%2C%22inversion%22%3A2%2C%22playability%22%3A5%2C%22extras%22%3A3%7D%2C%7B%22notes%22%3A%5B7%2C7%2C6%2C9%2C9%2C7%5D%2C%22inversion%22%3A3%2C%22playability%22%3A5%2C%22extras%22%3A3%7D%2C%7B%22notes%22%3A%5B12%2C11%2C9%2C9%2C12%2C12%5D%2C%22inversion%22%3A1%2C%22playability%22%3A5%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B16%2C14%2C14%2C13%2C12%2C12%5D%2C%22inversion%22%3A2%2C%22playability%22%3A5%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B16%2C14%2C14%2C16%2C17%2C16%5D%2C%22inversion%22%3A2%2C%22playability%22%3A5%2C%22extras%22%3A3%7D%2C%7B%22notes%22%3A%5B19%2C19%2C18%2C21%2C21%2C19%5D%2C%22inversion%22%3A3%2C%22playability%22%3A5%2C%22extras%22%3A3%7D%2C%7B%22notes%22%3A%5B7%2C7%2C6%2C4%2C5%2C7%5D%2C%22inversion%22%3A3%2C%22playability%22%3A7%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B19%2C19%2C18%2C16%2C17%2C19%5D%2C%22inversion%22%3A3%2C%22playability%22%3A7%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B4%2C2%2C2%2C1%2C0%2C4%5D%2C%22inversion%22%3A2%2C%22playability%22%3A8%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B4%2C2%2C2%2C1%2C5%2C4%5D%2C%22inversion%22%3A2%2C%22playability%22%3A8%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B4%2C2%2C6%2C4%2C5%2C4%5D%2C%22inversion%22%3A2%2C%22playability%22%3A8%2C%22extras%22%3A3%7D%2C%7B%22notes%22%3A%5B4%2C7%2C6%2C4%2C5%2C7%5D%2C%22inversion%22%3A2%2C%22playability%22%3A8%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B16%2C14%2C14%2C13%2C12%2C16%5D%2C%22inversion%22%3A2%2C%22playability%22%3A8%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B16%2C14%2C14%2C13%2C17%2C16%5D%2C%22inversion%22%3A2%2C%22playability%22%3A8%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B16%2C14%2C18%2C16%2C17%2C16%5D%2C%22inversion%22%3A2%2C%22playability%22%3A8%2C%22extras%22%3A3%7D%2C%7B%22notes%22%3A%5B16%2C19%2C18%2C16%2C17%2C19%5D%2C%22inversion%22%3A2%2C%22playability%22%3A8%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B0%2C2%2C2%2C1%2C0%2C4%5D%2C%22inversion%22%3A1%2C%22playability%22%3A9%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B12%2C11%2C9%2C13%2C9%2C12%5D%2C%22inversion%22%3A1%2C%22playability%22%3A9%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B12%2C14%2C14%2C13%2C12%2C16%5D%2C%22inversion%22%3A1%2C%22playability%22%3A9%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B7%2C7%2C6%2C9%2C5%2C7%5D%2C%22inversion%22%3A3%2C%22playability%22%3A12%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B19%2C19%2C18%2C21%2C17%2C19%5D%2C%22inversion%22%3A3%2C%22playability%22%3A12%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B0%2C2%2C2%2C4%2C0%2C4%5D%2C%22inversion%22%3A1%2C%22playability%22%3A14%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B12%2C14%2C14%2C16%2C12%2C16%5D%2C%22inversion%22%3A1%2C%22playability%22%3A14%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B4%2C2%2C2%2C4%2C0%2C4%5D%2C%22inversion%22%3A2%2C%22playability%22%3A15%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B12%2C11%2C9%2C13%2C12%2C12%5D%2C%22inversion%22%3A1%2C%22playability%22%3A15%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B16%2C14%2C14%2C16%2C12%2C16%5D%2C%22inversion%22%3A2%2C%22playability%22%3A15%2C%22extras%22%3A0%7D%2C%7B%22notes%22%3A%5B4%2C2%2C2%2C4%2C0%2C0%5D%2C%22inversion%22%3A2%2C%22playability%22%3A18%2C%22extras%22%3A1%7D%2C%7B%22notes%22%3A%5B16%2C14%2C14%2C16%2C12%2C12%5D%2C%22inversion%22%3A2%2C%22playability%22%3A18%2C%22extras%22%3A1%7D%5D%0Aconst%20newGroups%20%3D%20R.splitEvery%2812%2C%20chords%29%3B%0Aconsole.log%28newGroups%29